### ISSUE TITLE
Report websites where Browserpass doesn't work

### ISSUE THREAD

**Issue by maximbaz** [browserpass/browserpass-extension#62]
Please use this thread to report websites where Browserpass doesn't fully work, e.g. Browserpass is unable to fill username or password, or fills wrong form, or autosubmit doesn't work.

I'm not promising to support every single website, but I will investigate every example and if possible improve the detection logic.

I will not however add special cases for certain websites, I don't want Browserpass become a collection of broken websites on the Internet 🙂

**Comment by 532910** [comment]
Autosubmit doesn't work for:
https://esia.gosuslugi.ru/
Discourse sites:
https://community.opencellid.org
https://community.letsencrypt.org
https://forum.f-droid.org

**Comment by maximbaz** [comment]
Discourse sites will unfortunately not work until they improve their markup, sadly they are not very interested in doing so: https://github.com/discourse/discourse/pull/7056

**Comment by 532910** [comment]
https://ticket.coreboot.org/login
Filed wrong: login goes into openid url, login field leaved unchanged

**Comment by maximbaz** [comment]
esia.gosuslugi.ru works for me, make sure to test in the latest version 😉

**Comment by 532910** [comment]
I use СНИЛС login

**Comment by 532910** [comment]
version 3.0.3

**Comment by maximbaz** [comment]
I don't have the right credentials so I can't really test a successful login, but I click on СНИЛС, fill it with random credentials and I see the "Войти" button becoming grey, something is refreshing, and then I get "wrong username or password" red error, so as far as I can tell auto-submit actually works? 😕 What do you see exactly? And what browser do you use?

**Comment by 532910** [comment]
firefox 66.0.1-1, debian sid
it work for test record with random СНИЛС and password (I see "Введено неверное имя пользователя или пароль")
but for my credentials it just doesn't press "Войти" button

**Comment by maximbaz** [comment]
I have the same behavior for random СНИЛС and password, seems like I won't be able to reproduce because I don't have the correct credentials 😞 
Injection is a bit annoying to debug, but I'll give it another try a bit later.

By the way, sadly there will be cases when autosubmit only works in Chromium but not in Firefox, because we have now an additional way of submitting forms, but Firefox doesn't support this (yet?): https://github.com/browserpass/browserpass-extension/pull/55

It's still an improvement, because in v2 such forms didn't work in both Firefox and Chromium 🙂

**Comment by 532910** [comment]
can reproduce for random credentials:
```
% pass gosuslugi.ru    
hftEv*Iv\DrNMiB+&uf<!$sd}NWRlp
user: 123-123-512 35
```
for is not submitted, no red alert "Введено неверное имя пользователя или пароль"

**Comment by 532910** [comment]
https://ikus.pesc.ru/login autologin doesn't work

**Comment by 532910** [comment]
sorry my fault, please ignore message about pesc.ru

**Comment by maximbaz** [comment]
> ticket.coreboot.org/login
> Filed wrong: login goes into openid url, login field leaved unchanged

Fixed by #70

**Comment by maximbaz** [comment]
I tried `esia.gosuslugi.ru` and I can't reproduce the issue, neither in Chromium, nor in Firefox.

My exact steps:

1. Create the following password entry (includes `autoSubmit` for simpler repro steps):

```
hftEv*Iv\DrNMiB+&uf<!$sd}NWRlp
user: 123-123-512 35
autoSubmit: yes
```

2. Load latest Browserpass
3. Open incognito window, navigate to esia.gosuslugi.ru
4. Click СНИЛС
6. Use Browserpass

I do get a red alert about incorrect password.

**Comment by 532910** [comment]
both firefox 66.0.1-1 and chromium 73.0.3683.75-1 with clean profiles are able to submit (get incorrect password error) with fake credentials that I specified, sorry.

but both do not submit my real credentials

**Comment by 532910** [comment]
Looks like I've found it, try this one:
```
% pass gosuslugi.ru
oe`7\5IIxJ<_Rj2eYmiRs?U_"(+@Z.
user: 134–345–234 65
q: fBp8lbkYP9B82WivrxWM
a: H7wKupXuiwx0MPmhyF2M
```

**Comment by maximbaz** [comment]
This turned out to be a BEAUTIFUL discovery, a bug in a most unexpected location. Very happy that you gave me precise repro steps! Solved in #87 🙂

**Comment by OkanEsen** [comment]
`pi.hole` doesn't seem to work and doesn't return any password. If I remove the prefilled domain and type it manually into the search bar, the extension does seem to find the password.

![browserpass-1](https://user-images.githubusercontent.com/552226/56082819-c2fe0280-5e1d-11e9-9d0a-eb2e7e76071a.png)
![browserpass-2](https://user-images.githubusercontent.com/552226/56082820-c2fe0280-5e1d-11e9-9f07-fc5cd4dcaa6a.png)

**Comment by maximbaz** [comment]
This is an interesting case @OkanEsen. The reason for this behavior is because Browserpass is trying to determine where a domain name is in the string `Privat/Logins/pi.hole`, and it does so by using Tldjs, which knows how to search for a valid real well-known domain. In this case, `.hole` is not a real TLD, therefore Browserpass fails to detect this particular password entry.

@OkanEsen were the screenshots made on `https://pi.hole` domain, or on some other domain?

@erayd what do you think about teaching `pathToDomain` to also take into consideration the current domain from address bar? In other words, if you locally server a website on `https://pi.hole` and you have entries like `pi.hole.gpg` or `login.pi.hole.gpg`, they should appear in popup even though `.hole` is not a known TLD?

**Comment by OkanEsen** [comment]
> @OkanEsen were the screenshots made on `https://pi.hole` domain, or on some other domain?

Yes, that's correct and the default domain for a PiHole installation afaik.

It's not a huge dealbreaker for me, though I should mention, that there are other devices in my network, which are using non standard domains too, such as `https://fritz.box` etc.

**Comment by maximbaz** [comment]
I think it's also fairly common to host local servers on non-existent domains for local development, so I also tend to vote for doing this, unless anyone can think of any security issues with doing that.

**Comment by erayd** [comment]
> ...to also take into consideration the current domain from address bar?

I think this is a good idea. It'll remember after the first use anyway, so it's not a big deal, but I think this will be a useful improvement to the first-use UX.

**Comment by lewisdiamond** [comment]
newegg.ca doesn't work:
`Error: TypeError: Cannot read property 'filledFields' of null`

**Comment by maximbaz** [comment]
Nice one, will be fixed by https://github.com/browserpass/browserpass-extension/pull/110, thanks for reporting!

**Comment by TychoSchenkeveld** [comment]
Autosubmit doesn't work on my work's Identity Provider (PingID) with the new version. With the old (2.x) it did work. Tested both on Chrome and Firefox, on Mac. Autosubmit is switched on and works on other sites.

It's hard to give a direct link as it's only meant to be called from another site. But if you go to https://spiceportal.se.com you should be redirected to it. You won't be able to get in, but you should be able to see the form.

**Comment by mithodin** [comment]
https://ebay-kleinanzeigen.de also has `Error: TypeError: Cannot read property 'filledFields' of null`

**Comment by maximbaz** [comment]
OK I'll prioritize releasing a bugfix release 👍

**Comment by maximbaz** [comment]
`3.0.11` is out with the fix for the TypeError above.

**Comment by maximbaz** [comment]
@TychoSchenkeveld I checked the website you posted above, sadly that website is no longer supported unless you can get ahold of its developers and convince them to improve HTML markup.

![image](https://user-images.githubusercontent.com/1177900/56141438-164a8f00-5f9d-11e9-8f4a-5979e436831e.png)

Currently the "Sign on" button is not even a `<button>` (let alone `[type=submit]` as it should be), but is also not identifiable as a _login_ button, it has no ID or class or name that would say something like "login" (title is usually localized, thus we don't take it into account).

The reason it worked in v2 is because back then we also attempted to submit the form itself, however unfortunately this action broke many popular websites like AWS (see https://github.com/browserpass/browserpass-legacy/issues/217#issuecomment-429671923).

I am surprised that auto-submit doesn't work in Chromium though, because v3 is also attempting to trigger and Enter keypress, which does submit the form on AWS for example, but for some reason has no effect on this page. Not sure yet if this is something we can improve, or the website is trying to detect and ignore fake Enter keypresses...

**Comment by TychoSchenkeveld** [comment]
> @TychoSchenkeveld I checked the website you posted above, sadly that website is no longer supported unless you can get ahold of its developers and convince them to improve HTML markup.

Thanks, I was afraid of that... I know this site isn't the best. I don't think I can convince them to change it. It's a big identity service, we're not the only ones using it.

> The reason it worked in v2 is because back then we also attempted to submit the form itself, however unfortunately this action broke many popular websites like AWS (see [browserpass/browserpass-legacy#217 (comment)](https://github.com/browserpass/browserpass-legacy/issues/217#issuecomment-429671923)).

Ah I see, I think this was also why it didn't work with Steam, it would submit there but not actually login. Also with Arstechnica.

> I am surprised that auto-submit doesn't work in Chromium though, because v3 is also attempting to trigger and Enter keypress, which does submit the form on AWS for example, but for some reason has no effect on this page. Not sure yet if this is something we can improve, or the website is trying to detect and ignore fake Enter keypresses...

Ah too bad, I was going to suggest that. I didn't realise you were already trying that. Yes indeed, it's very strange, I already noticed that when I press enter myself it works. It could very well be doing this. Too bad, I use this site a lot. It's supposed to use SSO (Kerberos) tickets but on Mac this doesn't work so it challenges me with logins every time. But you're right, their form leaves a lot to be desired.

**Comment by meskes** [comment]
ox.credativ.com works nicely on firefox but with chromium it hangs on "Filling login details..."

**Comment by maximbaz** [comment]
This is an interesting example, it is caused by a bug in Chromium, [others have stumbled upon it too](https://stackoverflow.com/questions/34406330/asbygoogle-causes-chrome-executescript-never-to-return).

I have found a workaround to circumvent such websites. You might notice that the filling will be a bit slow on this website, it is because the workaround is simply an "abort" call if "Filling login details" hangs for too long, but it's better than nothing.

Will be part of the next release.

**Comment by meskes** [comment]
Thanks a lot. What causes this? There does not seem to be an adsbygoogle block on that website.

**Comment by maximbaz** [comment]
I'm not exactly sure what causes it, but the symptoms are reproducible 100%.

Here's the iframe that causes Chromium to hang on `ox.credativ.com`:

![image](https://user-images.githubusercontent.com/1177900/56392694-65f5b880-6232-11e9-88f1-f45a3cd5bc28.png)

Maybe it's because it's hidden, or because its source is set to `javascript:false`, it's hard to say for sure. Sadly we can't decide which frames we want or don't want to inject into, it's all or nothing, but hopefully the workaround I have in mind is sufficient and won't cause troubles for anyone else.

**Comment by DamienCassou** [comment]
https://launchpad.37signals.com/signin (more details at https://github.com/browserpass/browserpass-extension/issues/134)

**Comment by 532910** [comment]
https://login.aliexpress.com/
sometimes it works, sometime it says: `Error: Error: Unable to inject script in the top frame`

**Comment by maximbaz** [comment]
I can't repro the error on Aliexpress, could you identify what is different when the fill fails? Different browsers? Different logins?

**Comment by 532910** [comment]
same browser, same login, different tries

**Comment by maximbaz** [comment]
In this case it could simply mean injection is happening very slow on your computer, we have a timeout set to `1000 ms`, but it was set almost arbitrarily - could you play with this constant and tell me if setting it to larger value helps, as well as what is the smallest value where you can't reproduce the issue anymore?

https://github.com/browserpass/browserpass-extension/blob/6c1208b77a502da313dae98638775ec242eb6e2a/src/background.js#L332

**Comment by sapolsky** [comment]
khanacademy.org - fills only password but not username

If I understand correctly, the site doesn't use proper HTML markup on purpose (safety?). They are not the only ones.

for khanacademy.org "Email or username" field:
form class="_123zd8nNaN"
input class="_2a3kdy"
placeholder="" type="text"
id="uid-identity-text-field-2-email-or-username"
data-test-id="identifier"
value=""
data-_extension-text-contrast=""

I know you don't want to add hacks for every such site. But perhaps [feature request] we could add funny field names to our pass-store files like we do with "login: adam.smith@whatever.com". In case of khanacademy - "uid-identity-text-field-2-email-or-username: <password>"

**Comment by erayd** [comment]
@maximbaz I remember we talked this over a while back and decided against custom selectors in password files. I don't remember why though - do you?

**Comment by maximbaz** [comment]
> khanacademy.org - fills only password but not username

Thanks for reporting!

> If I understand correctly, the site doesn't use proper HTML markup on purpose (safety?).

There's no safety in using improper HTML markup 🙂 In this particular case, they put login and password fields in different `<form>`s, which is very unusual - we intentionally don't fill other forms if we found a form that contains password field, to prevent accidentally leaking your credentials.

If you feel strongly about this, I recommend contacting them via support and asking to make their login form more password-manager friendly, feel free to include me in the discussion if it is public, I'd be happy to assist them if necessary, clarify questions, etc.

> They are not the only ones.

Please report them all in this ticket 😉 Even though we might not be able to support all of them, we do learn and improve based on some examples people post here!

> But perhaps [feature request] we could add funny field names to our pass-store files like we do with "login: adam.smith@whatever.com". In case of khanacademy - "uid-identity-text-field-2-email-or-username: "

> I remember we talked this over a while back and decided against custom selectors in password files. I don't remember why though - do you?

Yeah such feature request already exists (#74), we haven't posted yay or nay in there yet, but it is true that we have discussed it, and tending against supporting that feature. Apart from a few non-compliant websites, people also want to use that feature for different reason, e.g. to support "account" field (as on AWS) or credit card fields; and in all cases it feels wrong to force each user to repeat the same work and come up with selectors to different websites (which are identical for everyone on AWS for the account field, but at the same time different for credit cards on every single website), and to maintain those selectors when websites change. I think we will be able to come up with better solutions to all these scenarios than implementing custom selectors.

But anyway, let's leave this feature request aside for the moment, I'll start the discussion in #74 after I cleanup my schedule a little bit, for now you can subscribe to that ticket, contact khanacademy and post here more examples where autofill doesn't work as expected so I can have a look (if you want) 😉 Thanks!

**Comment by 532910** [comment]
```Error: Error: Unable to inject script in the top frame``` on aliexpress happens when firefox shows page loading indicator. If I stop loading with Esc and ask Browserpass to fill credentials it works fine.

**Comment by erayd** [comment]
@532910 That sounds like you're trying to fill the login prior to the page being ready. If you let the page finish loading, it should work correctly.

Is there a reason you are trying to fill the login before the page has loaded?

**Comment by 532910** [comment]
As sometimes complete page loading takes so long as I even can't say will it be ever loaded. (I suppose this is aliexpress issues or my flashblock.)
But page looks loaded (except progress indicator) when I press Browserpass button.

**Comment by 532910** [comment]
Is it possible to automatically stop page loading when I press Browserpass button?

**Comment by erayd** [comment]
@532910 Gotcha, that makes sense.

> Is it possible to automatically stop page loading when I press Browserpass button?

I'll need to do some investigating, but my suspicion is that this will be a bad idea. Some sites still use long-polling for fetching data rather than more modern methods (e.g. websockets), and if we kill the page load when browserpass is triggered, we'll also kill those connections, which is problematic.

**Comment by 532910** [comment]
https://accounts.firefox.com/signin
```Error: Error: Unable to inject script in the top frame```

**Comment by maximbaz** [comment]
Regarding the "Unable to inject script in the top frame" error, have you been able to try out [my suggestion to increase the wait constant](https://github.com/browserpass/browserpass-extension/issues/62#issuecomment-496199333)? We can try to make it better by increasing the value if it proves to help, but in general I think there's no better solution, we either have to hang Browserpass and wait until browser finishes loading the page (potentially minutes), or do what we do now and cancel the attempt on timeout... 🤔

**Comment by savyajha** [comment]
The website sbicard.com doesn't work with Browserpass. If I fill in the details manually and hit enter, it works, but if browserpass does the filling it tells me the user/pass combo is wrong.

By the name of the site, I hope that you understand my inability to provide you with a username/password for testing purposes. :)

**Comment by maximbaz** [comment]
On the first glance I don't see an obvious place where the issue might be, let's double check some details:

- Is this the login page you use? https://www.sbicard.com/creditcards/app/user/login
- Do you have autoSubmit enabled?
- What happens if Browserpass fills the form, but you manually press "Login" button?

**Comment by 532910** [comment]
> Regarding the "Unable to inject script in the top frame" error, have you been able to try out [my suggestion to increase the wait constant](https://github.com/browserpass/browserpass-extension/issues/62#issuecomment-496199333)?

Sorry, no. Is it possible to load browserpass-extension without makeing it? (I'd not like to install yarn and node.) Could you try to fill https://accounts.firefox.com/signin with any credentials. This issue is reproducable on firefox 67.0.4 with even a clean profile. (Chromium works fine.)

**Comment by maximbaz** [comment]
@532910 I have found an answer for that particular case: Firefox is blocking extensions from injecting any scripts on their domains:

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts

```
Note that content scripts are blocked on the following domains:

accounts-static.cdn.mozilla.net
accounts.firefox.com               <----------
addons.cdn.mozilla.net
addons.mozilla.org
api.accounts.firefox.com
content.cdn.mozilla.net
content.cdn.mozilla.net
discovery.addons.mozilla.org
input.mozilla.org
install.mozilla.org
oauth.accounts.firefox.com
profile.accounts.firefox.com
support.mozilla.org
sync.services.mozilla.com
testpilot.firefox.com
```

Seems like we can't fix that particular use-case 😞 

However, if you can reproduce this issue on other websites (like aliexpress.com), I would still be interested if you could play with the value of `MAX_WAIT`. You don't need to install any tools on your computer, here's what you need to do:

1. Download [browserpass-firefox-3.1.1.zip](https://github.com/browserpass/browserpass-extension/releases/download/3.1.1/browserpass-firefox-3.1.1.zip) from Github releases
2. Unarchive somewhere on your computer
3. In Firefox, go to `about:debugging#addons`
4. Click "Load temporary addon" and select `manifest.json` from the unpacked folder
5. Then in `background.dist.js`, find `MAX_WAIT` and change its value to `1500`, `2000`, etc.
6. After every change, save file, in `about:debugging#addons` click on "Reload" button next to Browserpass, then perform your test

**Comment by savyajha** [comment]
> On the first glance I don't see an obvious place where the issue might be, let's double check some details:
> 
>     * Is this the login page you use? https://www.sbicard.com/creditcards/app/user/login
> 
>     * Do you have autoSubmit enabled?
> 
>     * What happens if Browserpass fills the form, but you manually press "Login" button?

Yes, that is the login page that I use. I don't use auto-submit. I press enter manually after filling. Also, sorry for the late reply.

**Comment by maximbaz** [comment]
I can't find anything wrong with sbicard.com, but it's difficult to be sure because I don't have a real account to test successful login flow, I try with fake credentials and receive "bad credentials".

Is it possible that your password entry for sbicard.com contains a typo, that it differs from what you are typing manually?

If that's not the case, sadly I'm not able to proceed on this particular example 😞

**Comment by erayd** [comment]
@maximbaz I took a look at it, and it seems there are event handlers attached to the password field which might be messing with the entered value when typed, but not when pasted or filled through the DOM (which is what we are doing).

@savyajha Does that page work correctly if you paste the username and password (instead of having browserpass fill them), and then press enter to submit?

**Comment by 532910** [comment]
> Firefox is blocking extensions from injecting any scripts on their domains

Is it hardcoded into Firefox? If so you should also hardcode the list of mozilla domains and show explanation, instead of the error.



I can reproduce this issue on aliexpress.com and `MAX_WAIT` will not help. As sometimes aliexpress never ends to load the page and show loading progress permanently. It may be related to my ABP or third party cookies blocking, I don't know.


> You don't need to install any tools on your computer, here's what you need to do:

Maxim, thank you for clear explanation!

**Comment by savyajha** [comment]
@erayd It works perfectly if I copy from the terminal using gopass. However, if I use the browser extension to copy the password it doesn't work.

**Comment by erayd** [comment]
@savyajha That sounds like unknown weird behavior.

Which browser (and which version of that browser) are you using?

Which version of Browserpass (both extension and native host app) are you using?

**Comment by savyajha** [comment]
@erayd I'm using Firefox 67.0.4 on OSX. I'm using v3.1.1 of the Browserpass extension and v3.0.6 of the native host as downloaded and installed using Homebrew.

**Comment by eknoes** [comment]
Doesnt work on vBulletin Board: https://community.contao.org/en/

**Comment by maximbaz** [comment]
Thanks for the report! I've checked and it doesn't work because the site is doing something weird, they hid the password field and placed a text-field on top of it, this login form is designed against good practices and we don't try to fill out hidden fields or random text fields with passwords for security reasons.

![image](https://user-images.githubusercontent.com/1177900/62897606-f01edb00-bd53-11e9-8129-38614d505136.png)

If you feel strongly about it, I recommend you to contact developers and ask them to improve their HTML markup, maybe there's a chance they would fix it.

**Comment by eknoes** [comment]
Thanks for taking a look, I'll do that!

**Comment by 532910** [comment]
Just for the note: aliexpress works significant better now.

**Comment by r7l** [comment]
https://signin.rockstargames.com does not work. It fills out the password but not the email field.

**Comment by maximbaz** [comment]
@r7l thanks for this example, it was a really difficult one but I managed to fix it in #171 🙂

**Comment by r7l** [comment]
Thanks allot!

**Comment by jeLee6gi** [comment]
Neither field is getting filled on https://epicgames.com/id/login

**Comment by maximbaz** [comment]
Thanks for the report 👍 So far can't crack this one, it seems they have custom javascript events that clear the field after Browserpass types in credentials 😕 Will keep this in mind in case a bright idea visits me on how to approach this.

**Comment by maximbaz** [comment]
@jeLee6gi: @erayd has fixed this, it is part of `3.4.0` release 😉

**Comment by meskes** [comment]
I received word that https://meine.santander.de/login/#/login/ does not work.

**Comment by maximbaz** [comment]
Thanks! This one will probably stay unsupported, the website shoots itself in the foot with their javascript logic...

Even if I make some compromises I can get Browserpass to fill the form, but the login button stays disabled (because their javascript code considers the form to be empty), and then the only option for a user is to click in the password field, which clears the field anyway.

![login](https://user-images.githubusercontent.com/1177900/67151244-81b24800-f2c3-11e9-9a42-781a18d88adf.gif)

**Comment by jeLee6gi** [comment]
> @jeLee6gi: @erayd has fixed this, it is part of `3.4.0` release

Thank you @erayd and @maximbaz, that was super fast and it works like a charm

**Comment by 532910** [comment]
Doesn't work on https://login.aliexpress.com more:
`Error: Error: No fillable forms available for fields: login, secret`

**Comment by maximbaz** [comment]
@532910 this page has a login form in iframe in another domain (`passport.aliexpress.com`), and the error you see basically means Browserpass asked you if you allow it to fill the form in an iframe and you said "no".

We plan to implement a UI to allow users to change their mind, but it has been a very low priority so far. For now, go to Browserpass options and click "Clear usage data" button, and then try to login again :)

**Comment by 532910** [comment]
Thank you for explanation. Works fine now!

**Comment by 532910** [comment]
https://grabr.io/en/: the form is filled, but not submitted automatically.

**Comment by erayd** [comment]
@maximbaz This looks like a failure of the "use the debugger to press enter" logic.

1. Form fill is handled correctly (works)
2. Cannot unambiguously determine the login button, so don't autosubmit that way (works)
3. Focus the password field (works)
4. Attach the debugger and press enter (doesn't work)

There is an error emitted *after* the keypress code is completed: "Detached while handling command". Not detaching the debugger resolves this (albeit not in a production-suitable manner), but the form still isn't submitted correctly. I'm not particularly familiar with the `chrome.debugging` API - are you able to take a look at this?

**Comment by erayd** [comment]
@maximbaz Issue with https://www.barnesandnoble.com/ - it fills correctly, but throws an error when (manually) submitted. However, this error is only present when Browserpass has filled the form; it's fine otherwise. I've not tested with autosubmit.

![image](https://user-images.githubusercontent.com/121486/67736095-33433d00-fa6b-11e9-891b-97590292a05c.png)

**Comment by 532910** [comment]
https://www.inf-it.com/infcloud/
the form is filled, but not submitted automatically

**Comment by erayd** [comment]
@532910 

> https://www.inf-it.com/infcloud/
> the form is filled, but not submitted automatically

I have tested this and cannot reproduce it. When autosubmit is enabled, the form is submitted automatically, and seems to be behaving entirely correctly. Could you please provide instructions to reproduce the failure?

**Comment by 532910** [comment]
Just tested on the clean profile, and it still doesn't work. Yes, autosubmit is enabled. (I've checked is on another site.) It just fills the form and doesn't submit it.

```
% pass inf-it.com
test
user: test@inf-it.com
```

Browserpass 3.4.1, Firefox 70.0 (64-bit), Debian sid

**Comment by maximbaz** [comment]
`inf-it.com` is an example where autosubmit works only in Chromium (thanks to [debugger permission](https://github.com/browserpass/browserpass-extension#requested-permissions)). I'll look into the other cases now, thanks for submitting so many 😉

**Comment by 532910** [comment]
BTW, can (or will) Firefox Lockwise help with integration or it's just another internal password manager that can't be extended?

**Comment by erayd** [comment]
@532910 It's just another password manager - as far as I can see, there is no API component that would allow us to use it for managing form fills / submits.

**Comment by maximbaz** [comment]
> grabr.io/en: the form is filled, but not submitted automatically.

@532910 fixed in #188 but only in Chromium, it's again the same story, their login button isn't properly marked as a button that will submit form, so we put the focus inside the form, and in Chromium we are able to issue `Enter` keypress, but not in Firefox.

> @maximbaz Issue with barnesandnoble.com - it fills correctly, but throws an error when (manually) submitted. However, this error is only present when Browserpass has filled the form; it's fine otherwise.

@erayd I can't reproduce, I've created an account, in Chromium in incognito I open this page, click "Sign in", press `Ctrl+F` and `Enter` and I get logged in...

**Comment by erayd** [comment]
@maximbaz I've tracked down the cause.

I have a 16 character password. The password field on B&N has `maxlength="15"` on it, which means that typed passwords are truncated. However, Browserpass sets the value directly, and doesn't truncate - hence the error.

Do you mind if I open a PR that will truncate filled passwords if maxlength is present?

**Comment by maximbaz** [comment]
Wow, interesting, please go ahead, I think it's something we must have in the code :)

**Comment by erayd** [comment]
@maximbaz #189 :-)

**Comment by DamienCassou** [comment]
On [liberapay](https://liberapay.com/sign-in), the password is correctly filled-in but not the username.

**Comment by maximbaz** [comment]
Thanks, good one, fixed in https://github.com/browserpass/browserpass-extension/commit/c92beaf387e39266f79716328314388e47104f03

**Comment by DamienCassou** [comment]
Maxim Baz <notifications@github.com> writes:
> Thanks, good one, fixed in https://github.com/browserpass/browserpass-extension/commit/c92beaf387e39266f79716328314388e47104f03

Thank you

-- 
Damien Cassou

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill

**Comment by tomclark** [comment]
Hi,

I was going to raise a feature request to ask if it'd be possible to add a context menu option so that when you right-click a form field, you can tell Browserpass what to enter in that field. It seems like that might help with this issue where sites don't correctly mark up their pages and Browserpass doesn't know what to do.

The login page for the Home Assistant control panel is guilty of this and Browserpass doesn't know which fields to fill, giving the error:

```Error: Error: No fillable forms available for fields: login, secret```

The Blur plugin does the kind of thing I mean, where you right-click a field and the choose what to enter in that field (I have no accounts on this page because obviously I don't use Blur for passwords):

![image](https://user-images.githubusercontent.com/784012/68889281-862c1e00-0714-11ea-89bf-7e1b7f3674f1.png)

Would that be possible? i.e. add a Browserpass option in the browser's context menu and allow you to choose which pass field to enter?

I'm using SRWare Iron 75 on macOS Catalina, FWIW, but I guess it could be made to work for all Chrome-based browsers. Apologies if this has been addressed before. I searched for "right-click" and "context" but couldn't find anything obvious.

**Comment by 532910** [comment]
1. A separate issue should be opened.
2. The continuation:
> can tell Browserpass what to enter in that field
and remember the decision to fill it automatically next time

**Comment by maximbaz** [comment]
Thanks for the suggestion, will track in #195

**Comment by akskap** [comment]
on [onlinesbi](https://retail.onlinesbi.com/retail/login.htm), username is not filled in correctly but password is

**Comment by zchlm** [comment]
[Soundcloud](https://soundcloud.com) - iframe popup is dismissed on password entry

**Comment by RalfJung** [comment]
Browserpass does not seem to work on old reddit: when I want to log in via https://old.reddit.com/r/uBlockOrigin/, it is unable to fill in my reddit credentials.

**Comment by maximbaz** [comment]
Haven't visited this issue for a while, thanks for your reports guys!

-------

onlinesbi.com - interesting edge case, fixed

----

Soundcloud is an interesting example, it doesn't fill anything because the form is inside an iframe, on a different origin, and with `allow-modals` disabled it prevents Browserpass from asking a user to confirm whether it is okay to submit this form or not. @erayd any ideas if we can do anything about it?

![image](https://user-images.githubusercontent.com/1177900/90042048-7143e180-dcca-11ea-8076-456b90b4d58d.png)

----

Old reddit - we don't follow the cursor position when deciding which form to fill, there is a different logic in place. Given that on reddit both forms are valid and browserpass _technically_ picked a valid login form, I tend to avoid doing anything else in this case 😅

**Comment by RalfJung** [comment]
> Old reddit - we don't follow the cursor position when deciding which form to fill, there is a different logic in place. Given that on reddit both forms are valid and browserpass technically picked a valid login form, I tend to avoid doing anything else in this case 

The form it picked is kind of hard to see at that point, because it is shaded out. But your call, I'll not use the popup any more now that I discovered the existence of that other login form.^^

**Comment by 532910** [comment]
https://www.ozon.ru/

**Comment by 532910** [comment]
https://www.itead.cc/

**Comment by 532910** [comment]
https://www.ozon.ru/

I've added it and removed later as it doesn't requires password for login, though username (e-mail/phone) is filled wrong (username is put into the search field.)

**Comment by RalfJung** [comment]
On https://www.doncamillo-saarbruecken.de/, filling in the password only works after I clicked into the password field once to clear its contents.

**Comment by maximbaz** [comment]
Thanks as usual 🙂 

> https://www.itead.cc/
> https://www.ozon.ru/

These two show similar behavior but I currently have no good ideas how they could be solved without breaking a lot of other websites, their markup violates our core assumption that if website developers know what a `<form>` tag is, they would _at least_ use it for login form - so we have some fallback logic that just takes any available form, and on these websites it turns out to be the search form... Unless I come up with something clever, they will have to be supported by https://github.com/browserpass/browserpass-extension/issues/195

> https://www.doncamillo-saarbruecken.de/

This one is so silly we can't support it, they change the field _type_ on click via javascript, so until a user clicks on the field it is truly a non-interesting regular text field where passwords most certainly should not be inserted... 😄

**Comment by 532910** [comment]
https://bill.nevalink.net
login and password are filled, but not submitted

**Comment by 532910** [comment]
https://zadarma.com/en/
Error: Error: No fillable forms available for fields: login, secret

**Comment by 532910** [comment]
https://www.voip-info.org
login and password are filled, but not submitted

**Comment by equaeghe** [comment]
https://orcid.org/signin

Password is filled, but username is not.
* Username field:

   ```
   <input _ngcontent-xov-c94="" matinput="" formcontrolname="username" apptrim="" class="mat-input-element mat-form-field-autofill-control ng-tns-c80-2 cdk-text-field-autofill-monitored ng-pristine ng-valid ng-touched" id="mat-input-0" aria-describedby="mat-hint-0" aria-invalid="false" aria-required="false">
   ```
* Password field:

  ```
  <input _ngcontent-xov-c94="" matinput="" formcontrolname="password" type="password" class="mat-input-element mat-form-field-autofill-control ng-tns-c80-3 cdk-text-field-autofill-monitored ng-pristine ng-valid ng-touched" id="mat-input-1" aria-invalid="false" aria-required="false">
  ```

**Comment by 532910** [comment]
RouterOS router webfig login page
login and password are filled, but not submitted

It could be tested online: http://demo.mt.lv/webfig/

**Comment by erayd** [comment]
Thanks @532910. Will investigate why this is, as fill + <key>enter</key> does work, so it seems odd that autosubmit doesn't.

As it happens, I use Browserpass to log in to RouterOS on a daily basis, and @maximbaz also has a router that runs it. So we don't need the online demo to test with. But I do appreciate that you linked it for us - thank you!

**Comment by erayd** [comment]
@532910 I've investigated, and it seems the RouterOS webfig login page doesn't actually contain a form at all. The fields are simply input fields in a table, and the login button isn't actually a button at all, but merely a styled link with a javascript click handler attached to it.

There's no way that Browserpass should ever be auto-submitting something like that sorry. However, autofill + pressing <key>enter</key> does work just fine.

**Comment by 532910** [comment]
Thank you for investigation

**Comment by DamienCassou** [comment]
It seems that browserpass can't find the submit button in https://app.nalo.fr/components/auth/#/login.

**Comment by morciej** [comment]
This is using Ver. **3.7.2** in Firefox **87.0**

Filling a modal html form works well for local addresses like `192.168.X.XXX:PORT/folder`. I make sure to name containing folder to `192.168.X.XXX:PORT`

However, **when I make a domain** for this server in `\etc\hosts` file (`192.168.X.XXX    server`) and then
name the folder above the .gpg to `server.PORT`, and the gpg file contains `url: server.:PORT/folder` then the form does not fill out. Trying to use Ctrl-Shift-F results in `Error: Error: Unable to inject script in the top frame`

I tried using `server.` or just `server` to be sure.

It would be nice to refer to my router as `router.` :-)

**Comment by morciej** [comment]
Again, Ver. 3.7.2 in Firefox 87.0
On https://www.fido.ca/profile/signin, browserpass fails to find any fillable elements. But when the site is displayed with no style, it finds them, and once reverted to basic page style, it keeps finding them (in the same session only).

**Comment by DamienCassou** [comment]
Doesn't work on: https://espacepro.cmb.fr/auth/login

**Comment by BastienDurel** [comment]
On proxmox hosts, 3.7.2/chromium : `Error: Error: Unable to inject script in the top frame`

For example : https://data-cloud4.data.fr:8006/
Works with Firefox

**Comment by maximbaz** [comment]
Thank you for posting these 👍

> - can't find the submit button in https://app.nalo.fr/components/auth/#/login.
> - https://www.fido.ca/profile/signin
> - https://www.airbnb.ca

The case here is that markup is weird, the fields which Browserpass doesn't find are simply not passing sanity check. In the first case it's unidentifiable login button (we avoid pressing a random button after putting credentials on the page), in the latter two cases it's just bad markup, login fields are actually made invisible with CSS and what you see on the page is divs drawn to look as fields... So browserpass doesn't treat those invisible fields as a real login form.

My best suggestion is to get those supported via https://github.com/browserpass/browserpass-extension/issues/195

> Filling a modal html form doesn't work for local addresses like 192.168.X.XXX

This I could not actually reproduce, or I misunderstood the issue. If I go to an IP address and it contains a html login form, and I have in my password store an entry called `<ip-addr>.gpg`, it will be discovered by Browserpass, and I would see it in popup.

If it's a modal HTTP form (a popup that browser shows, which prevents you from interacting with browser until you submit or cancel this popup), make sure to look at [this section of the docs](https://github.com/browserpass/browserpass-extension#modal-http-authentication), for example you must open such page from within Browserpass for it to be able to provide the credentials.

> Doesn't work on: https://espacepro.cmb.fr/auth/login

At the moment the root cause is https://github.com/browserpass/browserpass-extension/issues/73, didn't research if their markup is good otherwise.

**Comment by jbalme** [comment]
https://my.wealthsimple.com/app/login?locale=en-ca

**Comment by paulperegud** [comment]
https://www.centrum24.pl/

**Comment by aventrax** [comment]
@tomclark Home Assistant is really annoying for me. 

Any update? Home assistant has no "name" property in the input fields so I tried to add them but nothing changed. Where is the issue?

**Comment by 532910** [comment]
ikea.com
login and password are filled, but not submitted

**Comment by Sharpeee** [comment]
On
https://www.jollyroom.dk/mine-sider/log-ind
only "password" is completed, not "username". The "username" field is called "email" in the html.

**Comment by erayd** [comment]
On the login screen for https://www.ironwifi.com/, the password is auto-filled into both the username and password fields.

**Comment by 532910** [comment]
browserpass successfully fills username and password for openwrt's luci but doesn't automatically submit form after this

**Comment by 532910** [comment]
I can give my Home Assistant public address of for @maximbaz or @erayd for investigation, but would not like to publish it here.

**Comment by Hate9** [comment]
Sites using Authentik (https://goauthentik.io/) don't work.

**Comment by fuu0** [comment]
GLS Onlinebanking is broken:
https://www.gls-online-filiale.de/services_cloud/portal
only "password" is completed, not "username" - html Field for username is html element "vrNetKey"

**Comment by timonegk** [comment]
> Field for username is html element "vrNetKey"

"vrNetKey" is also used by other German banks, e.g. Volksbank and Sparda-Bank:
https://www.sparda-h.de/services_cloud/portal/login
https://www.sparda-bank-hamburg.de/services_cloud/portal/portal-oauth/login
https://www.hamburger-volksbank.de/services_cloud/portal
https://www.volksbank-eg.de/services_cloud/portal/
https://www.berliner-volksbank.de/services_cloud/portal/

**Comment by erayd** [comment]
Instagram (see #323)

**Comment by BastienDurel** [comment]
On 
https://www.borgbase.com/login

Error: Error: No fillable forms available for fields: login, secret

**Comment by foxyseta** [comment]
On https://www.interrail.eu/en/login#/login, the username field (email) is filled in properly, but the password is ignored. I could not see any output from the extension on the developer console. Am I supposed to be looking somewhere else?

**Comment by foxyseta** [comment]
Bump

**Comment by soren121** [comment]
Chase Bank (https://www.chase.com/) no longer works; I get this error message from Browserpass:

> Error: Error: No fillable forms available for fields: login, secret

**Comment by 532910** [comment]
> https://accounts.firefox.com/signin (The page is fully loaded) `Error: Error: Unable to inject script in the top frame`

the solition is to clean `extensions.webextensions.restrictedDomains` in `about:config`

**Comment by maximbaz** [comment]
Oh wow thanks for sharing, I didn't know it's possible to have any kind of workaround for mozilla's own domains!

**Comment by Tari-dev** [comment]
> > https://accounts.firefox.com/signin (The page is fully loaded) `Error: Error: Unable to inject script in the top frame`
> 
> the solition is to clean `extensions.webextensions.restrictedDomains` in `about:config`

Thanks for this, I even reinstalled the entire browserpass native thinking its an issue with it

**Comment by 532910** [comment]
Trying to fill login/password on BOG Business (https://bonline.bog.ge/login) gives several alerts and doesn't fill the form.

**Comment by 532910** [comment]
only password field is filled on https://iam.redsift.cloud/login

**Comment by maximbaz** [comment]
I suspect both [bonline.bog.ge/login](https://bonline.bog.ge/login), [reddit.com/r/rust](https://www.reddit.com/r/rust/) and https://secure.chase.com (not the chase.com homepage) are all the result of https://github.com/browserpass/browserpass-extension/issues/73! It's a shame we haven't got that solved, I'll add these examples to that issue, if someone feels inspired to try to solve this, the help is very welcome!

It looks like the devs of [iam.redsift.cloud/login](https://iam.redsift.cloud/login) tried really hard to conceal the "email" field, there's no indication in the markup that this could be a login field 🙄 I don't think we'll be able to do much about it, hopefully someone working there will fix the markup.

**Comment by alci63** [comment]
I hope this is still the place to report this kind of problems:

browserpass is not able to fill user / password in Authentik SSO portal https://goauthentik.io/

Shows "Error no fillable form available for fields: login, secret".

<img width="290" height="63" alt="Image" src="https://github.com/user-attachments/assets/aae1abc3-2799-45e1-b7da-0bf316f95e4f" />

Login field looks like this:
`<input name="uidField" autofocus="" autocomplete="username" spellcheck="false" class="pf-c-form-control" required="" aria-invalid="false" type="text" placeholder="Adresse email ou nom d'utilisateur" value="">`

Password field looks like this:
`<input autocomplete="current-password" required="" aria-invalid="false" type="password" id="ak-stage-password-input" name="password" placeholder="Veuillez saisir votre mot de passe" class=" pf-c-form-control pf-m-icon " value="">`

Regards
Franck

**Comment by dumain** [comment]
https://auth.lacon.org/ 
Shows:  "Error: Error: No fillable forms available for fields: login, secret"

As this is "Powered by authentik" probably the same issue as https://github.com/browserpass/browserpass-extension/issues/62#issuecomment-3126869250
